### PR TITLE
[NODERAWFS] Revert #18163

### PR DESCRIPTION
**Pull request by kleisauke**
This reverts commit c614fa57ea8048a34341891dbd6f99fb78b6869a.

---

Here's a standalone reproducer:

Before:
```console
$ curl -LO https://github.com/llvm/llvm-test-suite/raw/main/SingleSource/Benchmarks/Misc-C++/Large/ray.cpp
$ em++ -O3 -sEXIT_RUNTIME -sNODERAWFS ray.cpp -o ray.js
$ node ray.js > x.ppm
$ sha256sum x.ppm
acaa1db08ee30dafda03b41c77466b1435fe8ffdd4d88e5a5876e87dff126f58  x.ppm
$ vips copy x.ppm x.png
VipsImage: memory area too small --- should be 262144 bytes, you passed 79485
```

After:
```console
$ em++ -O3 -sEXIT_RUNTIME -sNODERAWFS ray.cpp -o ray.js
$ node ray.js > x.ppm
$ sha256sum x.ppm
cd6fa6fa75b4ba29077f7e4e531b96f17ff0b82f3c70df526c36f748d6a38c5c  x.ppm
$ vips copy x.ppm x.png
```
<details>
  <summary>x.png</summary>

x
</details>

### CODE REVIEW

File: test/other/test_fflush_fs_exit.out [hunk 1]
```diff
@@ -1,2 +1,2 @@
 Hey1
-Hey2
+Hey2
```
Comment (sbc100) [inline test/other/test_fflush_fs_exit.out:2]: Is this change needed?

**Comment by sbc100** [comment]
I'm inclined to perhaps revisit #18163.   It seems to me that if one uses `NODERAWFS` then overriding `stdout`, `stderr`, `stdin`, `print`, `printErr` should make just not effect the program stdout/stderr/stdin since the whole point of `NODERAWFS` is to directly map to node file I/O.

File: test/other/test_fflush_fs_exit.out (hunk 1 above)
Comment (kleisauke) [reply test/other/test_fflush_fs_exit.out:2]: This is needed to accommodate `other.test_fflush_fs_exit` under `NODERAWFS` since that can flush streams without newlines.

Comment (sbc100) [reply test/other/test_fflush_fs_exit.out:2]: I see that makes sense.

**Comment by kleisauke** [comment]
> I'm inclined to perhaps revisit #18163. It seems to me that if one uses `NODERAWFS` then overriding `stdout`, `stderr`, `stdin`, `print`, `printErr` should make just not effect the program stdout/stderr/stdin since the whole point of `NODERAWFS` is to directly map to node file I/O.

I agree, let's revert #18163. In retrospect, it seems that `library_tty.js` is very tied to the browser, as it skips null terminators and always flushes with a newline.

The latest revision of this PR reverts #18163 in a trivial way.

File: test/test_other.py [hunk 2]
```diff
@@ -9713,7 +9733,7 @@ def test_ioctl(self):
     # ioctl requires filesystem
     self.do_other_test('test_ioctl.c', emcc_args=['-sFORCE_FILESYSTEM'])
 
-  @also_with_noderawfs
+  # @also_with_noderawfs # NODERAWFS doesn't use library_tty.js for stdio streams
```
Comment (sbc100) [inline test/test_other.py:9736]: It seems that this should probably work under NODERAWFS.   Perhaps open a bug an refer it that bug here?

File: test/test_other.py [hunk 3]
```diff
@@ -1687,18 +1687,38 @@ def test_module_stdin(self):
...
+    assert(stderr.length === 0, 'stderr should be empty. \\n' +
+        'stderr: \\n' + stderr);
+    assert(UTF8ArrayToString(stdout, 0).startsWith('hello, world!'), 'stdout should start with the famous greeting. \\n' +
+        'stdout: \\n' + stdout);
+}
+''')
+    self.emcc_args += ['--pre-js', 'pre.js']
+    self.do_runf('hello_world.c')
+
+  # @also_with_noderawfs # NODERAWFS doesn't use library_tty.js for stdio streams
```
Comment (sbc100) [inline test/test_other.py:1710]: Just remove this line?

File: src/library_noderawfs.js [hunk 4]
```diff
@@ -55,6 +54,12 @@ addToLibrary({
       var mode = NODEFS.getMode(path);
       return { path, node: { id: st.ino, mode, node_ops: NODERAWFS, path }};
     },
+    createStandardStreams() {
+      FS.createStream({ nfd: 0, position: 0, path: '', flags: 0, tty: false, seekable: false }, 0);
```
Comment (kleisauke) [inline src/library_noderawfs.js:58]: Note that I set `tty: false` here to ensure that issue #22264 remains resolved.
```console
$ emcc ioctl_test.c -lnodefs.js -lnoderawfs.js -sEXIT_RUNTIME
$ node ./a.out.js
pc-kaw
a.out.js: /dev/stdin not a tty: Not a tty
```

**Review by sbc100 — APPROVED** [review]

File: src/library_noderawfs.js (hunk 4 above)
Comment (kleisauke) [reply src/library_noderawfs.js:58]: (I also set it for stdout and stderr, btw)

Comment (sbc100) [reply src/library_noderawfs.js:58]: Seems reasonable for now, but we should probably add a TODO there, and we should really be setting `tty` based on the underlying node `isatty` call for each descriptor.

Comment (kleisauke) [reply src/library_noderawfs.js]: Ah, I just noticed that `isatty` for stdio streams will return `0` and set `errno` to `ENOTTY` with this change, which is probably not what we want. I just pushed commit 88a5475f9e68be66d24d77ed974ecc65fbfa6947 instead. Let's reopen #22264 after this change.

File: test/test_other.py (hunk 2 above)
Comment (kleisauke) [reply test/test_other.py]: Changed to: `# @also_with_noderawfs # NODERAWFS reports ENOTTY, see issue #22264.` with commit faac3bc62820b63205890b91ac513b0c035d71b1 instead.

File: test/test_other.py (hunk 3 above)
Comment (kleisauke) [reply test/test_other.py]: Removed with commit faac3bc62820b63205890b91ac513b0c035d71b1.

File: test/test_other.py [hunk 5]
```diff
@@ -1687,18 +1687,38 @@ def test_module_stdin(self):
     self.emcc_args += ['--pre-js', 'pre.js']
     self.do_runf('module/test_stdin.c', 'hello, world!')
 
-  @also_with_noderawfs
+  # @also_with_noderawfs # NODERAWFS doesn't use library_tty.js for stdio streams
```
[the remark this answers is not in the document]
Comment (kleisauke) [reply test/test_other.py]: Removed with commit faac3bc62820b63205890b91ac513b0c035d71b1.

**Comment by kleisauke** [comment]
> Can you update the PR title to refer to the change you are reverting?

Sure, updated to `[NODERAWFS] Revert #18163`.
